Source/WebCore/style/values/box/StyleMargin.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "StyleMargin.h"
+
+#include "RenderStyleInlines.h"
+#include "StyleBuilderChecking.h"
+#include "StyleBuilderConverter.h"
+#include "StyleLengthWrapper+CSSValueConversion.h"
+
+namespace WebCore {
+namespace Style {
+
+auto CSSValueConversion<MarginEdge>::operator()(BuilderState& state, const CSSValue& value) -> MarginEdge
+{
+    auto cssToLengthConversionDataWithTextZoomFactorIfNeeded = [](BuilderState& state, bool isFontIndependentUnit) -> CSSToLengthConversionData {
+        auto zoom = state.cssToLengthConversionData().zoom();
+        auto usedZoom = 1.0f;
+
+        if (isFontIndependentUnit)
+            usedZoom = shouldUseEvaluationTimeZoom(state) ? 1.0f : state.style().usedZoom();
+        else
+            usedZoom = zoomWithTextZoomFactor(state);
+
+        if (usedZoom == zoom)
+            return state.cssToLengthConversionData();
```

Comment:
can you comment on why do we need this for MarginEdge?

Reply:
I copied this from Vitor's change, assuming it was a perf benefit (if our calculated zoom already matches the state's Length-adjusted zoom, just return it rather than adjusting to the same zoom.